### PR TITLE
Add tumor-normal calling mode to VarDictJavaEndToEnd

### PR DESCRIPTION
**Pull request by clintval**
No regressions or API change.

- [x] Add tumor-normal code paths (`test_somatic.R` and `var2vcf_paired.pl`)
- [x] Type all Pipes (`Text`/`Vcf`)
- [x] Added documentation (too much?)
- [x] Pulled out hard-coded values into defaults.
- [x] Ensure it works!

### CODE REVIEW

**Comment by clintval** [comment]
Ready for review. I confirm no regressions, the API is only extended.

I successfully ran this in tumour-only mode using an unchanged calling pipeline.

I also successfully ran this in tumour-normal mode and confirmed all new options work as expected.

File: tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala [hunk 1]
```diff
@@ -161,25 +248,38 @@ private class Var2VcfValid(sampleName: String,
...
 @clp(
   description =
     """
-      |Calls somatic variants in a tumor-only sample using VarDictJava.
+      |Call variants using `VarDictJava` in tumor-only or tumor-normal mode.
+      |
+      |`VarDictJava` is a variant discovery program written in Java. It is a sensitive variant caller for both single
+      |and paired sample variant calling from BAM files. VarDictJava implements several novel features such as amplicon
+      |bias aware variant calling from targeted sequencing experiments, rescue of long indels by realigning bwa
+      |soft-clipped reads, and better scalability than many other Java-based variant callers.
```
Comment (clintval) [inline tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala:268]: Copied from upstream. I can re-word or slim if you want.

File: tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala [hunk 2]
```diff
@@ -76,49 +98,113 @@ object VarDictJava extends Configuration {
...
+                          memory: Memory = VarDictJava.DefaultMemory
+                         ) extends ProcessTask with VariableResources with PipeOut[Text] {
   name = "VarDictJava"
 
   override def pickResources(resources: ResourceSet): Option[ResourceSet] = {
     resources.subset(minCores = Cores(minThreads), maxCores = Cores(maxThreads), memory = memory)
   }
 
+  /** Format tumor/normal BAM paths into one argument by joining on the default separator. Defaults to the tumor BAM path. */
+  private def maybeJoinTumorNormalPaths(tumorBam: PathToBam, normalBam: Option[PathToBam]): String = {
```
Comment (clintval) [inline tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala:127]: Instructions from here: https://github.com/AstraZeneca-NGS/VarDictJava#paired-variant-calling

**Review by nh13 — APPROVED** [review]
@clintval have you tested both tumor-only and tumor-normal modes?

**Comment by clintval** [comment]
@nh13, yes I confirm tumor-only invocations work (and are unchanged since there are no regressions here) and I also confirm that tumor-normal analysis works!

**Comment by nh13** [comment]
@tfenne care to take a look?

File: tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala [hunk 3]
```diff
@@ -76,49 +98,113 @@ object VarDictJava extends Configuration {
...
+    normalBam match {
+      case None             => tumorBam.toString
+      case Some(_normalBam) => Seq(tumorBam.toString, _normalBam.toString).mkString(VarDictJava.TumorNormalSeparator)
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala]: Not a big deal, but no need to `.toString` the paths before calling `.mkString`.

File: tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala [hunk 4]
```diff
@@ -76,49 +98,113 @@ object VarDictJava extends Configuration {
...
+                            minimumHighQualityAltDepth: Option[Int] = None,
+                            minimumAf: Double = VarDictJava.MinimumAf,
+                            minimumSignalToNoiseRatio: Option[Double] = None,
+                            minimumHomozygousAlleleFreq: Option[Double] = None
+                           ) extends ProcessTask with PipeWithNoResources[Text, Vcf] {
+  name = "Var2VcfPaired"
+
+  /** Format the two sample names into one argument by joining on the default separator. */
+  private def sampleNames: String = {
+    Seq(tumorName, normalName).mkString(VarDictJava.TumorNormalSeparator)
```
Comment (tfenne) [inline tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala:188]: Do you want to maybe add an assert here that neither sample name already contains `TumorNormalSeparator`?  I could see that causing havoc.

**Review by tfenne — APPROVED** [review]
A couple of minor comments from me.  Happy to merge this once you're ready @clintval

**Comment by clintval** [comment]
Thanks @tfenne, I added the validations and tested them out using my end-to-end tumor-only and tumor-normal samples.

All works as expected! Back to you.

**Pull request merged**